### PR TITLE
Twister Flashing Better Logging

### PR DESCRIPTION
**Pull request by ryanhagen**
While troubleshooting with twister flashing I found the logs to be useless. This adds some usefulness by:

1. Clearly specifying what output comes from the flash command to distinguish those logs from twister framework
2. Outputting the return code of the flash command
3. If twister cannot even start the flash command, tell us specifically

### CODE REVIEW

File: scripts/pylib/twister/twisterlib/handlers.py [hunk 1]
```diff
@@ -569,11 +569,14 @@ def handle(self):
                 try:
                     (stdout, stderr) = proc.communicate(timeout=60)
                     # ignore unencodable unicode chars
-                    logger.debug(stdout.decode(errors = "ignore"))
+                    decoded_stdout = stdout.decode(errors="ignore")
+                    decoded_stderr = stderr.decode(errors="ignore")
+                    logger.debug("Flash command stdout: %s", decoded_stdout)
+                    logger.error("Flash command stderr: %s", decoded_stderr)
```
Comment (golowanow) [inline scripts/pylib/twister/twisterlib/handlers.py:575]: stderr is already going to be written to `d_log` - see few lines below, isn't it ?

File: scripts/pylib/twister/twisterlib/handlers.py [hunk 2]
```diff
@@ -589,10 +592,10 @@ def handle(self):
             with open(d_log, "w") as dlog_fp:
                 dlog_fp.write(stderr.decode())
 
-        except subprocess.CalledProcessError:
+        except subprocess.CalledProcessError as call_error:
             halt_monitor_evt.set()
             self.instance.status = "error"
-            self.instance.reason = "Device issue (Flash error)"
+            self.instance.reason = f'Unable to open flashing process: {call_error.output.decode(errors="ignore")}'
```
Comment (golowanow) [inline scripts/pylib/twister/twisterlib/handlers.py:598]: i'd suggest to keep the old `Device issue (Flash error` + add more details from the exception

**Review by golowanow — CHANGES_REQUESTED** [review]
@ryanhagen, please squeeze your multiple commits here into one and follow the guidelines -
https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines

File: scripts/pylib/twister/twisterlib/handlers.py (hunk 1 above)
Comment (ryanhagen) [reply scripts/pylib/twister/twisterlib/handlers.py:575]: It doesn't make much sense to only output errors to a file. When i turn on twisters most verbose logging setting i should get errors on stdout. Even though this outputs to a file, its hard to find where the error begins and where it ends.

**Comment by kartben** [comment]
@ryanhagen not sure you wanted to close this PR?

**Comment by ryanhagen** [comment]
> @ryanhagen not sure you wanted to close this PR?

i opened a new one to follow the commit standards https://github.com/zephyrproject-rtos/zephyr/pull/57306

**Pull request closed without merging**